src/reports/models/database/cassandra/cassandraConnector.js

```diff
@@ -78,6 +80,21 @@ async function updateReport(testId, reportId, reportData) {
     return executeQuery(queryData.query, queryData.params, queryOptions);
 }
 
+async function deleteReport(testId, reportId) {
+    const reportToDelete = await executeQuery(GET_REPORT_SUMMARY, [testId, reportId], queryOptions);
+
+    const startTime = reportToDelete[0].start_time;
+    const startTimeDate = new Date(startTime);
+    const startTimeYear = startTimeDate.getFullYear();
+    const startTimeMonth = startTimeDate.getMonth() + 1;
+    const where = 'WHERE start_time_year=? AND start_time_month=? AND start_time=? AND test_id=? AND report_id=?';
+    const whereParams = [startTimeYear, startTimeMonth, startTime, testId, reportId];
+    const deleteLastReport = 'DELETE from last_reports';
+
+    await executeQuery(`${deleteLastReport} ${where}`, whereParams, queryOptions);
+    await executeQuery(DELETE_REPORT_SUMMARY, [testId, reportId], queryOptions);
```

Comment:
what do you say about deleting all the stats saved for the deleted report in table `reports_stats`?